src/models/drpc.ts

```diff
@@ -2,12 +2,24 @@ import z from 'zod'
 
 import { UUID } from './strings.js'
 
+const subjectIdParser = z.discriminatedUnion('idType', [
+  z.object({
+    idType: z.literal('product_and_quantity'),
```

Comment:
I guess with each query we will add another one with it's own specific content. E.g. provenance will contain it's own?

Reply:
correct